### PR TITLE
CI for FreeBSD

### PR DESCRIPTION
**Pull request by domenukk**
This enables Github Actions for FreeBSD

### CODE REVIEW

**Comment by domenukk** [comment]
@devnexen what do you think

**Comment by devnexen** [comment]
Looks nice, I also use rustup rather than the package version :-)

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -279,3 +279,22 @@ jobs:
...
+      id: test
+      uses: vmactions/freebsd-vm@v0
+      with:
+        usesh: true
+        prepare: |
+          pkg install -y curl
+          curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+        run: |
+          freebsd-version
```
Comment (devnexen) [inline .github/workflows/build_and_test.yml:298]: if you wish, you can install one of the llvm/clang version (the system clang does not have llvm dev headers) and points LLVM_CONFIG to /usr/local/bin/llvm-config-<version>

**Comment by devnexen** [comment]
you might need to install bash package too.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (domenukk) [reply .github/workflows/build_and_test.yml:303]: `/usr/local/bin/llvm-config-14` doesn't seem to work

Comment (domenukk) [reply .github/workflows/build_and_test.yml:303]: Ah, my `add` command may have been wrong

Comment (devnexen) [reply .github/workflows/build_and_test.yml:302]: ah add and install are aliases, sorry I realise I made a typo most likely :-) seems you found out

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -10,272 +10,304 @@ env:
...
+          . "$HOME/.cargo/env"
+          rustup toolchain install nightly
+          export LLVM_CONFIG=/usr/local/bin/llvm-config14
+          pwd
+          ls -lah
+          echo "local/bin"
+          ls -lah /usr/local/bin/
+          which llvm-config
```
Comment (devnexen) [inline .github/workflows/build_and_test.yml:310]: you might need to run `scripts/shmem_limits_fbsd.sh`

**Comment by domenukk** [comment]
A few linker/compiler warnings, probably it's not using our passes at all?
```
  clang-14: warning: argument unused during compilation: '-g' [-Wunused-command-line-argument]
  clang-14: warning: argument unused during compilation: '-O3' [-Wunused-command-line-argument]
  clang-14: warning: argument unused during compilation: '-funroll-loops' [-Wunused-command-line-argument]
  clang-14: warning: argument unused during compilation: '-D FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION=1' [-Wunused-command-line-argument]
  clang-14: warning: argument unused during compilation: '-fexperimental-new-pass-manager' [-Wunused-command-line-argument]
```

Else it seems fine

**Comment by devnexen** [comment]
that might be enough as first step wdyt ?

**Pull request merged**